libpod/runtime_img.go

```diff
@@ -132,3 +139,75 @@ func (r *Runtime) Build(ctx context.Context, options imagebuildah.BuildOptions,
 	_, _, err := imagebuildah.BuildDockerfiles(ctx, r.store, options, dockerfiles...)
 	return err
 }
+
+// Import is called as an intermediary to the image library Import
+func (r *Runtime) Import(ctx context.Context, source string, reference string, changes []string, history string, quiet bool) (string, error) {
+	var (
+		writer io.Writer
+		err    error
+	)
+
+	ic := v1.ImageConfig{}
+	if len(changes) > 0 {
+		ic, err = util.GetImageConfig(changes)
+		if err != nil {
+			return "", errors.Wrapf(err, "error adding config changes to image %q", source)
+		}
+	}
+
+	hist := []v1.History{
+		{Comment: history},
+	}
+
+	config := v1.Image{
+		Config:  ic,
+		History: hist,
+	}
+
+	writer = nil
+	if !quiet {
+		writer = os.Stderr
+	}
+
+	// if source is a url, download it and save to a temp file
+	u, err := url.ParseRequestURI(source)
+	if err == nil && u.Scheme != "" {
+		file, err := downloadFromURL(source)
+		if err != nil {
+			return "", err
+		}
+		defer os.Remove(file)
+		source = file
+	}
+
+	newImage, err := r.imageRuntime.Import(ctx, source, reference, writer, image.SigningOptions{}, config)
+	if err != nil {
+		return "", err
+	}
+	return newImage.ID(), nil
+}
+
+// donwloadFromURL downloads an image in the format "https:/example.com/myimage.tar"
+// and temporarily saves in it /var/tmp/importxyz, which is deleted after the image is imported
+func downloadFromURL(source string) (string, error) {
+	fmt.Printf("Downloading from %q\n", source)
+
+	outFile, err := ioutil.TempFile("/var/tmp", "import")
+	if err != nil {
+		return "", errors.Wrap(err, "error creating file")
```

Comment:
Error should include the file name so I know for debugging what failed.

Reply:
i can add debug file, but i think if TempFile fails, you wont know the name